newlib/libc/include/sys/pgmspace.h

```diff
@@ -0,0 +1,47 @@
+/* PGMSPACE.H - Accessor utilities/types for accessing PROGMEM data */
+
+#ifndef _PGMSPACE_H_
+#define _PGMSPACE_H_
+
+#undef ICACHE_RODATA_ATTR
```

Comment:
Is it ok for these to be unconditionally #undef? Why not something like this:
```
#ifndef ICACHE_RODATA_ATTR
#define ICACHE_RODATA_ATTR
#endif
```

Reply:
Didn't think of that, but absolutely yes in the libc/include/sys/pgmspace.h (dummy, called on non-ESP8266 archs).

In the sys/xtensa/include/pgmspace.h, though, the PSTR() macro needs to be hard-set to include the 4-byte alignment (really, really speeds up operations at the cost of 1.5 bytes/string).

I'll tweak the default one...